src/main/java/org/jboss/aerogear/keycloak/metrics/MetricsEventListener.java

```diff
@@ -14,40 +14,26 @@
     @Override
     public void onEvent(Event event) {
         logEventDetails(event);
-
-        switch (event.getType()) {
-            case LOGIN:
-                PrometheusExporter.instance().recordLogin(event);
-                break;
-            case REGISTER:
-                PrometheusExporter.instance().recordRegistration(event);
-                break;
-            case LOGIN_ERROR:
-                PrometheusExporter.instance().recordLoginError(event);
-                break;
-            default:
-                PrometheusExporter.instance().recordGenericEvent(event);
-        }
+        PrometheusExporter.instance().recordEvent(event);
     }
 
     @Override
     public void onEvent(AdminEvent event, boolean includeRepresentation) {
         logAdminEventDetails(event);
-
-        PrometheusExporter.instance().recordGenericAdminEvent(event);
+        PrometheusExporter.instance().recordAdminEvent(event);
     }
 
     private void logEventDetails(Event event) {
         logger.infof("Received user event of type %s in realm %s",
-                event.getType().name(),
-                event.getRealmId());
+            (event.getType() == null ? "" : event.getType().name()),
```

Comment:
I think we should not obscure null names here, it's just a log and it might be more helpful to detect nullity.

Reply:
@josemigallas do you mean instead of an empty string we should insert something like `type is null` when the type is null?

Reply:
No, I mean I would leave it just like it was, printing `type null`.

Reply:
But if the event type is null then `event.getType().name()` would raise a NPE

Reply:
@josemigallas +1 on printing null explicitly. I will do that.
However, when using null as a label value, Prometheus client generates an exception saying "cannot use null as label value". So, I will keep that other part the same.

Reply:
Ah! you're right I misread the condition. Then yes, I would prefer `type is null` as you said.